pkg/ginkgo_helper/metrics.go

```diff
@@ -0,0 +1,113 @@
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := api.NewClient(api.Config{
+		Address: url.String(),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return v1.NewAPI(client), nil
+}
+
+type PromQLInstance struct {
+	description string
+	query       string
+	api         *v1.API
+}
+
+func PromQL(description string, query string, api *v1.API) PromQLInstance {
+	return PromQLInstance{
+		description: description,
+		query:       query,
+		api:         api,
+	}
+}
+
+type promQLEvaluatedMatcher struct {
+	evaluated   bool
+	actual      model.Value
+	description string
+	promQL      string
+}
+
+func PromQLEvaluatedToEmpty() *promQLEvaluatedMatcher {
+	return &promQLEvaluatedMatcher{}
+}
+
+func (matcher *promQLEvaluatedMatcher) Match(actual interface{}) (success bool, err error) {
+	ql, ok := actual.(PromQLInstance)
+	if !ok {
+		return false, fmt.Errorf("PromQLEvaluatedToEmpty must be passed a ginkgo_helper.PromQLInstance. Got:\n%s", format.Object(actual, 1))
+	}
+
+	result, value, err := matcher.checkPromQL(*ql.api, ql.query, time.Now())
+	matcher.evaluated = err != nil
+	matcher.actual = value
+	matcher.promQL = ql.query
+	matcher.description = ql.description
+	return result, err
+}
+
+// checkPromQL copied from github.com/PingCAP-QE/metrics-checker with val returned
+// checkPromQL checks if a query returns true.
+func (matcher *promQLEvaluatedMatcher) checkPromQL(client v1.API, query string, ts time.Time) (result bool, value model.Value, err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	val, _, err := client.Query(ctx, query, ts)
+	if err != nil {
+		return false, nil, err
+	}
+	// Ref: https://github.com/prometheus/prometheus/blob/76750d2a96df54226e85ac272d7ad5a547630240/rules/manager.go#L186-L206
+	switch v := val.(type) {
+	case model.Vector:
+		return v.Len() > 0, val, nil
+	case *model.Scalar:
+		return v != nil, val, nil
+	default:
+		return false, val, errors.New("rule result is not a vector or scalar")
+	}
+}
+
+func (matcher *promQLEvaluatedMatcher) FailureMessage(_ interface{}) (message string) {
+	if !matcher.evaluated {
+		return "PromQLInstance has not evaluated yet."
+	} else {
+		return format.Message(matcher.actual, fmt.Sprintf("to match the expected result of PromQL [%s] %s", matcher.description, matcher.promQL), "<non-empty Scalar or Vector>")
+	}
+}
+
+func (matcher *promQLEvaluatedMatcher) NegatedFailureMessage(_ interface{}) (message string) {
+	if !matcher.evaluated {
+		return "PromQLInstance has not evaluated yet."
+	} else {
+		return format.Message(matcher.actual, fmt.Sprintf("to match the expected result of PromQL [%s] %s", matcher.description, matcher.promQL), "<empty Scalar or Vector>")
+	}
+}
+
+func init() {
+	gomega.SetDefaultConsistentlyDuration(3 * time.Hour)
+	gomega.SetDefaultConsistentlyPollingInterval(1 * time.Minute)
+	gomega.SetDefaultEventuallyTimeout(3 * time.Hour)
+	gomega.SetDefaultEventuallyPollingInterval(1 * time.Minute)
```

Comment:
Why change default values here?

Reply:
default value is "`Eventually` will poll every 10 milliseconds for up to 1 second and `Consistently` will monitor every 10 milliseconds for up to 100 milliseconds" which is undoubtedly too frequent for a metrics checking, and the timeout is too short as we need to moniter the status of a cluster in a relatively long time as well.

Reply:
Can we only set default value for metrics-checking? It seems that this PR changes the  global default value.

Reply:
seems not supportted by gomega. Or we could make our own version of `Consistently` (maybe `ConsistentlyOnMetrics`...?) that works as a delegate of the original `Consistently` with longer interval & duration passed in.

Reply:
Maybe we can define some varaibles and case can use them in metrics matcher?